src/main/java/com/google/cloud/spanner/jdbc/JdbcConnection.java

```diff
@@ -96,6 +111,19 @@ static boolean useLegacyValidCheck() {
     return false;
   }
 
+  @VisibleForTesting
+  static Attributes createMetricAttributes(DatabaseId databaseId) {
```

Comment:
Can we add a `private final` field to `JdbcConnection` that contains these attributes, and only create it when the connection is created, instead of creating a new collection for every metric that we collect?

Reply:
Fixed.